File: restaurant/state/served_state.py
from state.dish_state import DishState

class ServedState(DishState):
    """Trạng thái: Đơn hàng đã được phục vụ"""
    
    def handle(self, context):
        """Xử lý khi đơn hàng được phục vụ cho khách"""
        print(f"Đơn hàng {context.order_id} đã được phục vụ.")
        context.state = ServedState()
    
    def get_status_name(self) -> str:
        return "Served"

File: backend/src/com/restaurant/pattern/state/ServedState.java
package com.restaurant.pattern.state;

/**
 * Trạng thái "Đã phục vụ" - Khi món ăn đã được phục vụ
 */
public class ServedState implements DishState {
    private OrderItem orderItem;

    public ServedState(OrderItem orderItem) {
        this.orderItem = orderItem;
    }

    @Override
    public void nextState() {
        throw new IllegalStateException("Món ăn đã được phục vụ, không thể thay đổi trạng thái");
    }

    @Override
    public String getStatus() {
        return "Đã phục vụ";
    }

    @Override
    public String toString() {
        return getStatus();
    }
}
